benchmarks/entropy_adiabat/plugins/entropy_advection.cc

```diff
@@ -169,6 +192,11 @@ namespace aspect
                                                  scratch.material_model_outputs,
                                                  scratch.heating_model_outputs);
 
+      // We compute the amount of diffusion based on the solution of the temperature equation.
+      std::vector<double> old_temperatures (n_q_points);
+      scratch.finite_element_values[this->introspection().extractors.temperature].get_function_values (this->get_old_solution(),
+          old_temperatures);
```

Comment:
here `scratch.old_temperature_values` should even be filled already with the correct values, no need to recompute (see entropy_viscosity.cc:467).